### ISSUE TITLE
Handle SQLite's incompatible change of implementation for LOG(x)

### ISSUE THREAD

**Issue by lukaseder** [jOOQ/jOOQ#15125]
`LOG(x)` was introduced as the SQLite implementation for `LN(x)`. But apparently, somewhere between 3.39.4 and 3.42.0, they changed this incompatibly and now `LOG(x)` is the same as `LOG10(x)`. I cannnot find any documented change in the release notes:
https://www.sqlite.org/changes.html

There is now a `LN(x)` function, which didn't exist until recently:
https://www.sqlite.org/lang_mathfunc.html#ln

See also:
- https://twitter.com/lukaseder/status/1662011336274894848

**Comment by lukaseder** [comment]
So, 3.40.0 introduced this change

```sql
select log(2), sqlite_version(), ln(2)
```

We get:

```
|log(2)      |sqlite_version()|ln(2)       |
|------------|----------------|------------|
|0.3010299957|3.40.0          |0.6931471806|
```

In 3.39.4, the `LN(2)` function didn't exist, and this query:

```sql
select log(2), sqlite_version()
```

Produces:

```
|log(2)      |sqlite_version()|
|------------|----------------|
|0.6931471806|3.39.4          |
```

**Comment by lukaseder** [comment]
I had only noticed because of this "forced" upgrade due to CVE-2023-32697 here: https://github.com/jOOQ/jOOQ/issues/15114.

**Comment by lukaseder** [comment]
Fixed for jOOQ 3.19.0. A backport isn't possible as we rely on a new `SQLDialect` version, and we can't force jOOQ Open Source users to upgrade in a patch release.

**Comment by lukaseder** [comment]
The support matrix has been updated to reflect this:
https://www.jooq.org/download/support-matrix

Users wishing to use SQLite 3.40+ with jOOQ are requested to work with `3.19.0-SNAPSHOT` for now, or avoid using `LN` with jOOQ.
